.githooks/pre-commit

```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+
+# This matches what is done in GitHub Actions (some of the checks are missing).
+
+rustup update stable
```

Comment:
How do you update to the most recent version?

Reply:
I use rustc from my system package manager so it can't be easily upgraded by a script. But I don't think the hook needs to perfrom the upgrade. Differences due to Rust versions aren't that frequent and the CI will catch it anyway.